test/reply-error.test.js

```diff
@@ -127,6 +128,58 @@ test('Should reply 400 on client error', t => {
   })
 })
 
+test('Should set the response from client error handler', t => {
+  t.plan(5)
+
+  function clientErrorHandler (err, socket, logger) {
+    t.type(err, Error)
+
+    const body = JSON.stringify({
+      error: 'Ended Request',
+      message: 'Serious Client Error',
+      statusCode: 400
+    })
+    logger.warn({ err }, 'Handled client error')
+    socket.end(`HTTP/1.1 400 Bad Request\r\nContent-Length: ${body.length}\r\nContent-Type: application/json; charset=utf-8\r\n\r\n${body}`)
+  }
+
+  const logStream = split(JSON.parse)
+  const fastify = Fastify({
+    clientErrorHandler,
+    logger: {
+      stream: logStream,
+      level: 'warn'
+    }
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    const client = net.connect(fastify.server.address().port)
+    client.end('oooops!')
+
+    var chunks = ''
+    client.on('data', chunk => {
+      chunks += chunk
+    })
+
+    client.once('end', () => {
+      const body = JSON.stringify({
+        error: 'Ended Request',
+        message: 'Serious Client Error',
+        statusCode: 400
+      })
+      t.equal(`HTTP/1.1 400 Bad Request\r\nContent-Length: ${body.length}\r\nContent-Type: application/json; charset=utf-8\r\n\r\n${body}`, chunks)
```

Comment:
I would put this string in a const and use it in the `clientErrorHandler` also since it is the same

Reply:
Done!